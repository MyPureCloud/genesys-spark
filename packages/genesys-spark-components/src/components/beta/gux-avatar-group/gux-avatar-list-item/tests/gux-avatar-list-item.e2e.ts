import { newSparkE2EPage, a11yCheck } from '../../../../../test/e2eTestUtils';

describe('gux-avatar-list-item', () => {
  describe('#render', () => {
    [
      `<div role="list"> <gux-avatar-list-item-beta><gux-avatar-beta name="John Doe"></gux-avatar-beta></gux-avatar-list-item-beta></div>`,
      `<div role="list"><gux-avatar-list-item-beta interactive-element="a"><gux-avatar-beta name="John Doe"></gux-avatar-beta></gux-avatar-list-item-beta></div>`,
      `<div role="list"><gux-avatar-list-item-beta focusable="true"><gux-avatar-beta name="John Doe"></gux-avatar-beta></gux-avatar-list-item-beta></div>`,
      `<div role="list"><gux-avatar-list-item-beta layout="plus-name"><gux-avatar-beta name="John Doe"></gux-avatar-beta></gux-avatar-list-item-beta></div>`
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-avatar-list-item-beta');

        await a11yCheck(page);

        expect(element.outerHTML).toMatchSnapshot();
      });
    });
  });
});
