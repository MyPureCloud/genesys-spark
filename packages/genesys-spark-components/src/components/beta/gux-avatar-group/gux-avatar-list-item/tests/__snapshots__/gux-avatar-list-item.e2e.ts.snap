// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-avatar-list-item #render should render component as expected (1) 1`] = `"<gux-avatar-list-item-beta role="listitem" hydrated=""><template shadowrootmode="open"><button type="button" tabindex="-1" class="gux-avatar-default"><slot></slot></button></template><gux-avatar-beta name="John Doe" hydrated=""></gux-avatar-beta></gux-avatar-list-item-beta>"`;

exports[`gux-avatar-list-item #render should render component as expected (2) 1`] = `"<gux-avatar-list-item-beta interactive-element="a" role="listitem" hydrated=""><template shadowrootmode="open"><a href="#" class="gux-avatar-default" tabindex="-1"><slot></slot></a></template><gux-avatar-beta name="John Doe" hydrated=""></gux-avatar-beta></gux-avatar-list-item-beta>"`;

exports[`gux-avatar-list-item #render should render component as expected (3) 1`] = `"<gux-avatar-list-item-beta focusable="true" role="listitem" hydrated=""><template shadowrootmode="open"><button type="button" tabindex="0" class="gux-avatar-default"><slot></slot></button></template><gux-avatar-beta name="John Doe" hydrated=""></gux-avatar-beta></gux-avatar-list-item-beta>"`;

exports[`gux-avatar-list-item #render should render component as expected (4) 1`] = `"<gux-avatar-list-item-beta layout="plus-name" role="listitem" hydrated=""><template shadowrootmode="open"><button type="button" tabindex="-1" class="gux-avatar-plus-name"><span class="gux-avatar-flex"><slot></slot><span aria-hidden="true">John Doe</span></span></button></template><gux-avatar-beta name="John Doe" hydrated=""></gux-avatar-beta></gux-avatar-list-item-beta>"`;
